src/controller/CHIPDeviceController.cpp

```diff
@@ -918,7 +918,14 @@ void DeviceCommissioner::OnSessionEstablishmentError(CHIP_ERROR err)
 {
     if (mPairingDelegate != nullptr)
     {
-        mPairingDelegate->OnStatusUpdate(DevicePairingDelegate::SecurePairingFailed);
+        if(err == CHIP_ERROR_PASSCODE_AUTHENTICATION_FAILED)
```

Comment:
I don't think this works.  See https://github.com/project-chip/connectedhomeip/issues/23913#issuecomment-1343502134

Was this tested?

Reply:
Ah, no, I was wrong; this comes up during handling of Pake2 on the client side.  So this does in fact work.

But the in-tree implementations of OnStatusUpdate need to be fixed to deal with the new value.  Right now at least `SetUpCodePairer::OnStatusUpdate` will misbehave (which would be a critical bug), and so might others.

The backwards-compatible way to do this would be to either send both status updates (invalid passcode _and_ failed) or communicate the failure reason in some other way.

Reply:
> If we send both the status updates as follows will it work?

I think you would probably want to send invalid passcode before failed if you send both....

But again, you should at least check the existing delegates and see what they do with that and whether they work right as a result.

Safer would be an entirely new notification.